### PR TITLE
feat(components): [color-picker] add `teleported` prop

### LINKED ISSUE #15357: [Component] [popover, color-picker] el-popover里面放一个el-color-picker，在el-popover里面打开取色器，在取色器中任意点击， el-popover就关闭了
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.6`
- Browser / OS: `谷歌`
- Build Tool: `Webpack`

### Related Component

- `el-popover`
- `el-color-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBzdHlsZT1cImRpc3BsYXk6IGZsZXg7IGFsaWduLWl0ZW1zOiBjZW50ZXJcIj5cbiAgICA8ZWwtcG9wb3ZlciBwbGFjZW1lbnQ9XCJyaWdodFwiIDp3aWR0aD1cIjQwMFwiIHRyaWdnZXI9XCJjbGlja1wiPlxuICAgICAgPHRlbXBsYXRlICNyZWZlcmVuY2U+XG4gICAgICAgIDxlbC1idXR0b24gc3R5bGU9XCJtYXJnaW4tcmlnaHQ6IDE2cHhcIj5DbGljayB0byBhY3RpdmF0ZTwvZWwtYnV0dG9uPlxuICAgICAgPC90ZW1wbGF0ZT5cbiAgICAgIDxlbC1jb2xvci1waWNrZXIgdi1tb2RlbD1cImNvbG9yMVwiIC8+XG4gICAgPC9lbC1wb3BvdmVyPlxuICA8L2Rpdj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IGNvbG9yMSA9IHJlZignIzQwOUVGRicpXG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

打开取色器，任意点击取色器，el-popover关闭

### What is Expected?

el-popover里面放一个el-color-picker，点击取色器，应该是完成取色后，el-popover也不会关闭

### What is actually happening?

el-popover里面放一个el-color-picker，打开取色器，在取色器里面任意点击el-popover就关闭了

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by zhixiaotong**
closed #15357

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by zhixiaotong** [comment]
之前有一个`pr`#15368 但已经太久了，过于粗暴，我看一直没人改，就自作主张改了。@kooriookami 麻烦看一下~

File: packages/components/color-picker/src/color-picker.ts [hunk 1]
```diff
@@ -52,6 +52,13 @@ export const colorPickerProps = buildProps({
     type: [String, Number],
     default: 0,
   },
+  /**
+   * @description whether color-picker popper is teleported to the body
+   */
+  teleported: {
+    type: Boolean,
+    default: true,
+  },
```
Comment (kooriookami) [inline packages/components/color-picker/src/color-picker.ts:61]:
```suggestion
  teleported: useTooltipContentProps.teleported,
```

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/color-picker/src/color-picker.ts (hunk 1 above)
Comment (zhixiaotong) [reply packages/components/color-picker/src/color-picker.ts:61]: ok

Comment (zhixiaotong) [reply packages/components/color-picker/src/color-picker.ts]: done

**Review by kooriookami — APPROVED** [review]
LGTM

**Pull request merged**